### ISSUE TITLE
Help block’s link text is too low contrast

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#10990]
### Issue Summary

The links within our help block’s warning, info, error variants have too light of a text color in light and dark mode. See for example the three versions in our styleguide: https://static-wagtail-v5-1.netlify.app/admin/styleguide/#help-section

Spotted as part of the upgrade notifier, which never shows in our test suite on `main` but does show when we work on pull requests for previous versions of Wagtail (#10989).

### Steps to Reproduce

1. Go to https://static-wagtail-v5-1.netlify.app/admin/styleguide/#help-section
2. Check the color contrast. It should be at least 4.5:1.

---

It looks like we could solve this changing the following styles: https://github.com/wagtail/wagtail/blob/main/client/scss/components/_help-block.scss#L18-L28

- Switch the base link color to `colors.secondary.400`
- Switch the hover color to `colors.secondary.600`.

This will match the link styles we had implemented for the "What’s new" banner (https://github.com/wagtail/wagtail/blame/main/client/scss/components/_whats-new.scss#L45), which use the same background as the "warning" help block.

**Comment by rohitsrma** [comment]
I'm working on this!!

**Comment by thibaudcolas** [comment]
Nice one @rohitsrma 🙌
